src/test/java/com/cegeka/academy/repository/UserChallengeTest.java

```diff
@@ -0,0 +1,90 @@
+package com.cegeka.academy.repository;
+
+import com.cegeka.academy.AcademyProjectApp;
+import com.cegeka.academy.config.audit.AuditEventConverter;
+import com.cegeka.academy.domain.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import javax.transaction.Transactional;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(classes = AcademyProjectApp.class)
+@Transactional
+public class UserChallengeTest {
+
+    @Autowired
+    UserChallengeRepository userChallengeRepository;
+
+    @Autowired
+    InvitationRepository invitationRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    ChallengeRepository challengeRepository;
+
+    Invitation invitation = new Invitation();
+    User user = new User();
+    Challenge challenge = new Challenge();
+
+    @Test
+    public void testAddUserChallenge() {
+
+        invitationRepository.save(invitation);
+
+        user.setLogin("LoginSetForTest");
+        user.setPassword("42jIG0vHCTEWClhT5R2om2V5NpgOXNcQggP6YJOz2xMccBQzGDWgqUDLKOqZ");
+        userRepository.save(user);
+
+        challenge.setStartDate(new Date());
+        challenge.setEndDate(new Date());
+        challenge.setStatus("Active");
+        challenge.setCreator(user);
+        challengeRepository.save(challenge);
+
+
+        UserChallenge userChallenge = new UserChallenge();
```

Comment:
Ai putea modulariza codul cu o metoda care sa iti dea o instanta dummy de `UserChallenge` in loc sa faci de 3 ori instantierea si setarea proprietatilor.